design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,174 @@
+
+The base table also determines the automatic joins that are performed when the user adds columns from linked tables.
+
+### Changing the base table
+
+Once the input table has been created, changing the base table will result in the loss of all progress. A warning should be given to users so that they understand the consequences of this change.
+
+### 1.1 Base Table Options
+
+The base table selector will display a list of all available tables in the current schema (Initially, this list will not include views that will be added in later iterations)
+
+To select a table, the user will find it on the list and click on the desired table option.
+
+Selecting the base table will enable the interface controls for adding additional steps, which were initially disabled. No action is available before a base table is selected.
+
+Once a selection has been made, the user will be prompted to proceed with the input table configuration, starting with column selection.
+
+---
+Wireframes
+
+[Selecting a Base Table](https://share.balsamiq.com/c/esUftomoFsZiRhZDMYdoPP.png)
+
+[Base Table Selected](https://share.balsamiq.com/c/me5CfGGeX3R35x2otLqbpu.png)
+
+## 2. Creating the Input Table
+
+The input table is created by selecting columns from the base table or creating new ones via formulas.
+
+These columns are added to the result table based on the selection order. Users can rename columns as they wish.
+
+By default, the system will name columns to indicate their source table and column according to the following [naming convention](#column-naming-convention).
+
+Some options might be available for columns that link to multiple records, like filters and aggregations to help users define which data they want to show and how. The system will automatically detect the type of columns and present the available options to users.
+
+### 2.1. Adding Input Columns
+
+To start adding columns, the user clicks on the `Add Column` button located in the `Columns` section on the query configuration panel. This action will display the `Column Selector` component in the inspector panel.
+
+The `Column Selector` will list all available columns, and those in use, when applicable. Additionally there will be a list of all formulas.
+
+The columns in `Column Selector` are listed in a hierarchical structure based on the links that exist between the base table and other tables, in both directions. Icons indicate whether the linked records are single or multiple.
+
+Links are presented as expandable sections containing the columns from the linked tables.
+
+When adding input columns, the user can add them using the list controls or drop the columns directly into the result table.
+
+---
+Wireframes
+
+[Adding Columns](https://share.balsamiq.com/c/oyxTXxqSh8rLqU3JY71DWY.png)
+[Added Column](https://share.balsamiq.com/c/7U9yahZtYWX2G5obkyivoz.png)
```

Comment:
We need to figure out how to represent the source's relationship to the base table (similar to how we do it in the column selection flow)